components/avatar/avatar.vue

```diff
@@ -102,36 +95,97 @@ export default {
       type: Object,
       default: () => ({}),
     },
+
+    /**
+     * Initials to be shown in the avatar. Used as fallback if image fails to load.
+     */
+    initials: {
+      type: String,
+      default: '',
+    },
   },
 
   data () {
     return {
       // initials, image or icon
-      kind: 'initials',
+      kind: 'image',
       AVATAR_SIZE_MODIFIERS,
-      AVATAR_COLOR_MODIFIERS,
       AVATAR_KIND_MODIFIERS,
       AVATAR_PRESENCE_SIZE_MODIFIERS,
+      imageLoadedSuccessfully: null,
+      slottedInitials: '',
+      formattedInitials: '',
     };
   },
 
+  computed: {
+    showDefaultSlot () {
+      return this.kind !== 'initials' && this.imageLoadedSuccessfully !== false;
+    },
+
+    showInitials () {
+      return this.kind === 'initials' || this.initials;
+    },
+
+    initialKindStyle () {
+      const randomGradientColorStops = this.randomizeGradientColorStops();
+      return {
+        '--avatar-gradient-angle': `${this.randomizeGradientAngle()}deg`,
+        '--avatar-gradient-stop-1': `var(--${randomGradientColorStops[0]})`,
+        '--avatar-gradient-stop-2': `var(--${randomGradientColorStops[1]})`,
+        '--avatar-gradient-stop-3': `var(--${randomGradientColorStops[2]})`,
+      };
+    },
+  },
+
   mounted () {
     this.init();
   },
 
   updated () {
-    this.init();
+    if (this.kind === 'initials') {
+      this.slottedInitials = this.$el.children[0].textContent;
+      this.formatInitials(this.slottedInitials);
+    }
   },
 
   methods: {
     init () {
-      const firstChild = this.$el.children[0];
+      const firstChild = this.$el.children[0] || this.$el;
```

Comment:
Key difference with the Vue2 version.
`this.$el.children[0]` was undefined when passing an initial in the slot so getting the root node element in that case and the text inside with `firstChild.textContent` in that case.

Reply:
yeah children doesn't work in vue 3 because the of the multi root change